### PR TITLE
setup pipeline

### PR DESCRIPTION
**Pull request by ShellyXueHan**
1. add the pipeline project
2. set build and deployment configs
3. set Jenkinsfile
4. update env vars in project

Please ignore the .pipeline folder as it's coming from another repo.
/bot-ignore-length

### CODE REVIEW

**Comment by ShellyXueHan** [comment]
@patricksimonian same here as the pr in reggie-web. Added in some functional updates together with the pipeline....which are 7d68885 (add refUrl as one more param) and b7274d6 (set cors to accept all base urls for web at the moment until we get better routing solution).

File: .pipeline/lib/build.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+const {OpenShiftClient, OpenShiftClientX} = require('pipeline-cli')
+const path = require('path');
+
+module.exports = (settings)=>{
+  const oc=new OpenShiftClientX({'namespace':'devhub-tools'});
```
Comment (patricksimonian) [inline .pipeline/lib/build.js]: oc = new

File: .pipeline/lib/deploy.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+const {Pipeline, OpenShiftClient, OpenShiftClientX} = require('pipeline-cli')
+const path = require('path');
+
+
+module.exports = (settings)=>{
+  const oc=new OpenShiftClientX();
```
Comment (patricksimonian) [inline .pipeline/lib/deploy.js]: oc = new

File: .pipeline/lib/deploy.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  // remove pr in prefix for test and prod environemnt:
+  const projectPrefix = oc.options.env === "dev" ? `-${oc.options.env}-${oc.options.pr}` : `-${oc.options.env}`
+
+  // set the rest of the env vars:
+  let extraParams = {
+    RM_HOST_VALUE: 'https://repo-mountie-devhub-prod.pathfinder.gov.bc.ca/bot/github/membership',
+    API_URL_VALUE: `https://${appName}${projectPrefix}-devhub-${oc.options.env}.pathfinder.gov.bc.ca`,
+  }
+
+  var objects = oc.process(oc.toFileUrl(templateFile), {
```
Comment (patricksimonian) [inline .pipeline/lib/deploy.js]: use `const` over `var`

File: .pipeline/lib/build.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  var templateFile = path.resolve(__dirname, '../../openshift/bc.yaml')
+  
+  const appName = 'reggie-api'
+
+  var objects = oc.process(oc.toFileUrl(templateFile), {
```
Comment (patricksimonian) [inline .pipeline/lib/build.js]: use `const` over `var`

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Jason Leach on 2018-01-10.
+//
+
+'use strict';
+
+const gulp = require('gulp');
```
Comment (patricksimonian) [inline gulpfile.js:23]: what is this for?

File: src/router/index.js [hunk 6]
```diff
@@ -21,12 +21,13 @@
...
 import cors from 'cors';
-import config from '../config';
+// import config from '../config';
 import ehlo from './routes/ehlo';
 import sso from './routes/ssoUsers';
 
 const corsOptions = {
-  origin: config.get('environment') === 'development' ? '*' : config.get('apiUrl'),
+  origin: '*',
+  // origin: config.get('environment') === 'development' ? '*' : config.get('apiUrl'),
```
Comment (patricksimonian) [inline src/router/index.js]: explain why commented?

**Review by patricksimonian — COMMENTED** [review]
looks good over all. some files need new line at end

File: gulpfile.js (hunk 5 above)
Comment (ShellyXueHan) [reply gulpfile.js:23]: for building the app. `"build": "NODE_ENV=production gulp"`

Comment (patricksimonian) [reply gulpfile.js:23]: ahh gotchya. I'd expect that the latest version of react scripts could do the build too. I'm sure there is something extra going on here is why gulp is being used.

**Comment by ShellyXueHan** [comment]
@patricksimonian ready!

**Pull request merged**